File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
class Solution {
    public int[] solution(int[] arr, int n) {
        int flag = 0;
        if(arr.length % 2 != 0) flag = 1;
        else flag = 0;
        for(int i = 0 ; i < arr.length; i++){
            if(flag == 1 && i % 2 == 0) arr[i] += n;
            else if(flag == 0 && i % 2 != 0) arr[i] += n;
        }
        return arr;
    }
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    if len(arr) % 2 == 1:
        return [arr[i] + n if i % 2 == 0 else arr[i] for i in range(len(arr)) ]
    else:
        return [arr[i] + n if i % 2 == 1 else arr[i] for i in range(len(arr)) ]